test/e2e/apimachinery/resource_quota.go

```diff
@@ -487,6 +488,61 @@ var _ = SIGDescribe("ResourceQuota", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 
+	It("should create a ResourceQuota and capture the life of a custom resource.", func() {
+		By("Creating a Custom Resource Definition")
+		testcrd, err := framework.CreateTestCRD(f)
+		Expect(err).NotTo(HaveOccurred())
+		defer testcrd.CleanUp()
+		countResourceName := "count/" + testcrd.Crd.Spec.Names.Plural + "." + testcrd.Crd.Spec.Group
+		time.Sleep(45 * time.Second)
```

Comment:
this looks suspect... what is this waiting for?

Reply:
Right after the CRD creation, the resource quota controller takes time to discover this new resource via discovery, and it is 30 seconds hardcoded here: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/core.go#L322.

If we do not wait for it here, let's say we create the quota immediately, now the resourcequota controller won't treat it as quotable yet, and treat it as synced successfully, no unless someone else trigger an update on the ResourceQuota object, it won't be queued anymore.

Though there is another loop which runs `enqueueAll` every 5 minutes to pick it up.

Reply:
ping @liggitt do you have suggestion on this?

Reply:
I'd suggest creating a test resourcequota object for the custom resource, and poll the quota object until the quota controller updates status (which indicates it discovered/resynced). Polling on a ~1 second interval means we'll continue the test as soon as possible, and giving it a longer timeout (at least 2x the quota controller sync period) means we'll be more robust to flakes.

Reply:
> I'd suggest creating a test resourcequota object for the custom resource, and poll the quota object until the quota controller updates status (which indicates it discovered/resynced)

@liggitt: This isn't going to work, as the question remains: when do we create the test resourcequota? it it is too fast, then the resource quota controller is gonna to skip it, and we end up with waiting for 5 minutes.

Reply:
oh, maybe you mean creating one test resourcequota object every second.

Reply:
we can create one, and keep updating it (set an annotation to an incrementing number or something) to keep it hitting the controller

Reply:
That's a good idea. 👍

Reply:
This is updated.

Reply:
Well, I can not update annotations, as resourcequota controller only enqueue the object when `spec.Hard` differs. I will switch to update `spec.Hard` instead.